src-node/index.ts

```diff
@@ -15,18 +15,61 @@ export interface FastFindInFiles {
   queryHits: QueryHit[]
 }
 
-const bindingWrapper = (directory: string, needle: string | RegExp): FastFindInFiles[] => {
-  if (!directory) {
-    throw new TypeError('Invalid input: Missing directory')
+interface FastFindInFilesOptions {
+  /**
+   * Absolute or relative directory path to search in.
+   */
+  directory: string
+  /**
+   * String or RegExp to search for.
+   */
+  needle: string | RegExp
+  /**
+   * Relative folder paths to exclude from the search.
+   *
+   * Requirements:
+   * - Must start with `./`
+   * - Must not end with `/`
+   */
+  excludeFolderPaths?: string[]
+}
+
+function fastFindInFiles(options: FastFindInFilesOptions): FastFindInFiles[] {
+  if (!options) {
+    throw new TypeError('Invalid input: Missing options')
   }
 
-  if (!needle) {
-    throw new TypeError('Invalid input: Missing needle')
+  if (typeof options.directory !== 'string' || options.directory.length === 0) {
+    throw new TypeError('Invalid input: Invalid or missing options.directory')
+  }
+
+  if ((typeof options.needle !== 'string' || options.needle.length === 0) && !(options.needle instanceof RegExp)) {
+    throw new TypeError('Invalid input: Invalid or missing options.needle')
+  }
+
+  if (options.excludeFolderPaths) {
+    if (!Array.isArray(options.excludeFolderPaths)) {
+      throw new TypeError('Invalid input: options.excludeFolderPaths must be an array')
+    }
+
+    options.excludeFolderPaths.forEach(excludeFolderPath => {
+      if (typeof excludeFolderPath !== 'string' || excludeFolderPath.length === 0) {
+        throw new TypeError('Invalid input: options.excludeFolderPaths.excludeFolderPath must be nonempty string')
+      }
+
+      if (excludeFolderPath.endsWith('/')) {
+        throw new TypeError('Invalid input: options.excludeFolderPaths.excludeFolderPath must not end with "/"')
+      }
+
+      if (!excludeFolderPath.startsWith('./')) {
+        throw new TypeError('Invalid input: options.excludeFolderPaths.excludeFolderPath must start with "./"')
+      }
```

Comment:
Hey @erikengervall, i think this code is wrong,
when i run fast-find-in-files not in the current cwd, this line prevents it (as it doesn't start with `./`.
when i change the code locally to ignore this check, it works,

why did we add this? what does that solve?

Reply:
also, what if the path starts with `../`?

Reply:
That's fair, I think I mixed  it up a bit with the trailing `/` restriction -- which did cause issues as it doesn't match with any folder path

We can remove this restriction entirely!

Reply:
https://github.com/erikengervall/fast-find-in-files/releases/tag/v1.0.1  removes the unnessary restriciton -- thanks for pointing it out!

Reply:
you rock 👑